### ISSUE TITLE
Add chart time range greater one year

### ISSUE THREAD

**Issue by shikousa** [openhab/openhab-webui#1905]
## The problem

For charts you can currently only select the time ranges DAY, WEEK, MONTH and YEAR.
If you select year you will see months in the charts.

![Link](https://community-openhab-org.s3.dualstack.eu-central-1.amazonaws.com/original/3X/a/3/a3f0472258490aedf21a216e77518685f77a3f32.png)

For energy visualisation it would be good to not only compare months, but also years.
So there should also be a chart time range like 3YEARS or even DECADE.

## Your suggestion

Add additional chart time ranges 3YEARS or even DECADE to visualize years

**Comment by Boldfor** [comment]
Adding these additional time ranges would not only be helpful to compare years, but also to check whether e.g. your energy bill is correct: A typical utilities company takes weeks to send you the invoice, with you not being able to go back, say, one year and a couple of weeks without having to use Grafana.

**Comment by marsic-84** [comment]
+ I want it too))  3,5,10 years, or at least 10y

**Comment by stefan-hoehn** [comment]
I added the 3-10 yrs option: #2190 

Does that fix that issue here?

**Comment by Boldfor** [comment]
@stefan-hoehn, thanks a lot.

Since loading times will be an issue (as mentioned here https://github.com/openhab/openhab-webui/pull/2190#pullrequestreview-1749430350) maybe it's a good idea to add 2Y and, if you want to keep the list short, remove the 10y option).

**Comment by stefan-hoehn** [comment]
I don't think we should remove then 10y option. I guess, the one who uses it should be just aware what they are doing. Just don't do it when you have lots of measurements taken on an item. Rather try to use aggregate items that only contain one entry per week or month by creating them via a special rule for example and then I am sure that the 10y option will work well (thats 52x10 = 520 values which is almost nothing to load)

**Comment by shikousa** [comment]
@stefan-hoehn Thanks for your effort!

You have extended this:
![grafik](https://github.com/openhab/openhab-webui/assets/3395690/3790607b-d137-4a13-9d41-e00764c105d6)

But this does not solve the issue here, as there should be also a decade option (or 3y, 5y) for charts.
![grafik](https://github.com/openhab/openhab-webui/assets/3395690/2cea4ecf-5873-4252-81ea-49dc8cdbd720)

In this post it is described how these charts can be created.
https://community.openhab.org/t/energy-charts-in-days-weeks-months-years/142488/12

As you already said, this is more effort to implement. Let's see who can take, but it would be very could if you could compare energy data year-by-year.

In terms of performance a decade should not be an issue if there is not too much.
For these charts we are using diff_first/diff_last anyway. At the moment Openhab takes all data, but this is not required. For a decade and using diff_first/diff_last, actually you only need to take 11 values from the database (first/last value of each year).

**Comment by stefan-hoehn** [comment]
Exactly, this is very I failed because it would require a new "magnitude" while my implementation just added a few instances more to an already existing, in my case Y(ear) which is much easier.
